src/mod_guildhouse_butler.cpp

```diff
@@ -208,25 +189,43 @@ class GuildHouseSpawner : public CreatureScript
             cost = GuildHouseTrainer;
             SpawnNPC(action, player);
             break;
-        case 2836:  // Blacksmithing
-        case 8128:  // Mining
-        case 8736:  // Engineering
-        case 18774: // Jewelcrafting (Alliance)
-        case 18751: // Jewelcrafting (Horde)
-        case 18773: // Enchanting (Alliance)
-        case 18753: // Enchanting (Horde)
-        case 30721: // Inscription (Alliance)
-        case 30722: // Inscription (Horde)
-        case 19187: // Leatherworking
-        case 19180: // Skinning
-        case 19052: // Alchemy
-        case 908:   // Herbalism
-        case 2627:  // Tailoring
-        case 19185: // Cooking
-        case 2834:  // Fishing
-        case 19184: // First Aid
+        case 28694: // Blacksmithing
+        case 28698: // Mining
+        case 28697: // Engineering
+        case 28701: // Jewelcrafting
+        case 28693: // Enchanting
+        case 28702: // Inscription
+        case 28700: // Leatherworking
+        case 28696: // Skinning
+        case 28703: // Alchemy
+        case 28704: // Herbalism
+        case 28699: // Tailoring
+        case 33587: // Cooking
+        case 28742: // Fishing
+        case 28706: // First Aid
+			cost = 0;
+
+			// Check if we need multiple trainers, a few require this otherwise
+			// you can't learn it soley in the guild house (Master is needed)
+			if (action == 28693)
+			{
+				// Enchanting
+				SpawnNPC(19540, player);
+			}
+			else if (action == 28700)
+			{
+				// Leatherworking
+				SpawnNPC(19187, player);
+			}
+			else if (action == 28703)
+			{
+				// Alchemy
+				SpawnNPC(19052, player);
+			}
+			
             cost = GuildHouseProf;
```

Comment:
```suggestion
        case 28706: // First Aid
            cost = 0;

            // Check if we need multiple trainers, a few require this otherwise
            // you can't learn it soley in the guild house (Master is needed)
            if (action == 28693)
            {
                // Enchanting
                SpawnNPC(19540, player);
            }
            else if (action == 28700)
            {
                // Leatherworking
                SpawnNPC(19187, player);
            }
            else if (action == 28703)
            {
                // Alchemy
                SpawnNPC(19052, player);
            }
            
            cost = GuildHouseProf;
```